include/s3select_functions.h

```diff
@@ -98,7 +102,10 @@ class s3select_functions
     {"#when-than#", s3select_func_En_t::WHEN_THAN},
     {"#case-when-else#", s3select_func_En_t::CASE_WHEN_ELSE},
     {"coalesce", s3select_func_En_t::COALESCE},
-    {"string", s3select_func_En_t::STRING}
+    {"string", s3select_func_En_t::STRING},
+    {"trim", s3select_func_En_t::TRIM},
```

Comment:
the following functions  "trim","leading","trailing"
expose (to user) the following generic function rule
trim(expression), leading(expression),trailing(expression)

it should be blocked, by adding "#trim#" and so on.